public/components/map_container/map_container.tsx

```diff
@@ -224,7 +224,14 @@ export const MapContainer = ({
 
   return (
     <div className="map-main">
-      {mounted && <MapsFooter map={maplibreRef.current!} zoom={zoom} />}
+      {mounted && maplibreRef.current && <MapsFooter map={maplibreRef.current} zoom={zoom} />}
```

Comment:
Because mounted will be set after maplibre instance ready, so do we still need check maplibreRef.current?
https://github.com/opensearch-project/dashboards-maps/blob/e569f114bb4fd31b6dbe969608068c73dd570615/public/components/map_container/map_container.tsx#L104

Reply:
Logically yes, but maplibreRef.current type includes undefined, but map props wants non-null. Another alternative is to use maplibreRef.current!, but, here, we are assuming state based on another state. It is always safe to check state to avoid any runtime failures in future.